code/Test_definitions/Geofencing.feature

```diff
@@ -0,0 +1,114 @@
+
+
+  @geofencing_subscriptions_03_Operation_to_retrieve_subscription_based_on_the_provided_ID
+  Scenario: Get a subscription based on provided id.
+    Given they use the geofencing url
+    When they get subscription for subscription-id
+    Then Response code is 200
+
+  @geofencing_subscriptions_04_Operation_to_delete_subscription_based_on_the_provided_ID
+  Scenario: Delete a subscription based on provided id.
+    Given they use the geofencing url
+    When they delete subscription for subscription-id
+    Then Response code is 204
+
+  @geofencing_subscriptions_05_Create_invalid_geofencing_subscription_for_a_device
+  Scenario:  Create geofencing subscription with invalid parameter
+    Given they use the geofencing url
+    When they create subscription with invalid parameter
+    Then Response code is 400
+
+  @geofencing_subscriptions_06_Get_invalid_geofencing_subscription_for_a_device
+  Scenario:  Get geofencing subscription with invalid subscription-id which is not available
+    Given they use the geofencing url
+    When they get subscription with invalid subscription-id which is not available
+    Then Response code is 404
+
+
+  @geofencing_subscriptions_07_Delete_invalid_geofencing_subscription_for_a_device
+  Scenario:  Delete geofencing subscription with invalid parameter
+    Given they use the geofencing url
+    When they delete subscription with invalid subscription-id
+    Then Response code is 404
+
+  @geofencing_subscriptions_08_Invalid_method_geofencing_subscription_for_a_device
+  Scenario:  Update geofencing subscription
+    Given they use the geofencing url
+    When they update subscription
+    Then Response code is 405
+
+
+  @geofencing_subscriptions_09_creation_of_subscription_for_subscribed_expired_time_in_past
+  Scenario:  Subscribed expire time in past for geofencing subscription
+    Given they use the geofencing url
+    When they create subscription with expire time in past
+    Then Response code is 400
+
+
+  @geofencing_subscriptions_10_creation_of_subscription_when_service_unavailable
+  Scenario: Subscription creation when service unavailable
+    Given they use the geofencing url
+    When they create subscription when service is unavailable
+    Then Response code is not  503
+
+  @geofencing_subscriptions_11_Get_Event-Details_of_subscription_entered
+  Scenario: Subscription creation when service have area-entered event
+    Given they use the geofencing url
+    When they create subscription with event have area-entered at "Boon"
+    When they create subscription with event have area-entered at "Berlin" and device enters "Berlin"
+    Then they get event details from notifications-url
+    Then Response code is 200
+
+  @geofencing_subscriptions_12_Get_Event-Details_of_subscription_left
+  Scenario: Subscription creation when service have area-left event
+    Given they use the geofencing url
+    When they create subscription with event have area-left with at "Boon"
```

Comment:
This has to be a test plan that may be run everywhere, we should avoid specific examples for specific locations